File: backend/models/Notification.js
const mongoose = require("mongoose");

const NotificationSchema = new mongoose.Schema(
  {
    type: {
      type: Number,
      // 0 = log, 1 = comment
      enum: [0, 1],
      required: true,
    },
    user: {
      type: mongoose.Schema.ObjectId,
      ref: "User",
      required: true,
    },
    task: {
      type: mongoose.Schema.ObjectId,
      ref: "Task",
      required: true,
    },
    unread: {
      type: Boolean,
      default: false,
    },
    workspace: {
      type: mongoose.Schema.ObjectId,
      ref: "Workspace",
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.model("Notification", NotificationSchema);

File: backend/app/models/notification.rb
# == Schema Information
#
# Table name: notifications
#
#  id                :bigint           not null, primary key
#  notification_type :integer
#  unread            :boolean
#  created_at        :datetime         not null
#  updated_at        :datetime         not null
#  task_id           :bigint
#  user_id           :bigint
#  workspace_id      :bigint
#
# Indexes
#
#  index_notifications_on_user_id  (user_id)
#
class Notification < ApplicationRecord
  belongs_to :task
  belongs_to :user
  belongs_to :workspace

  enum :notification_type => {
    :log_item => 0,
    :comment_item => 1
  }
end
